src/library/dao/directorydao.cpp

```diff
@@ -95,19 +97,22 @@ QList<RelocatedTrack> DirectoryDAO::relocateDirectory(
         return {};
     }
 
-    // on Windows the absolute path starts with the drive name
-    // we also need to check for that
-    QString startsWithOldFolder = SqlLikeWildcardEscaper::apply(
-        QDir(oldFolder).absolutePath() + "/", kSqlLikeMatchAll) + kSqlLikeMatchAll;
-
-    // Also update information in the track_locations table. This is where mixxx
-    // gets the location information for a track. Put marks around %1 so that
-    // this also works on windows
-    query.prepare(QString("SELECT library.id, track_locations.id, track_locations.location "
-                          "FROM library INNER JOIN track_locations ON "
-                          "track_locations.id = library.location WHERE "
-                          "track_locations.location LIKE '%1' ESCAPE '%2'")
-                  .arg(startsWithOldFolder, kSqlLikeMatchAll));
+    // Appending '/' is required to disambiguate files from parent
+    // directories, e.g. "a/b.mp3" and "a/b/c.mp3" where "a/b" would
+    // match both instead of only files in the parent directory "a/b/".
+    DEBUG_ASSERT(!oldFolder.endsWith('/'));
+    const QString oldFolderPrefix = oldFolder + '/';
+    const QString startsWithOldFolder = SqlLikeWildcardEscaper::apply(
+                                                oldFolderPrefix, kSqlLikeMatchAll) +
+            kSqlLikeMatchAll;
```

Comment:
This code is duplicated in `TrackDAO`. Should this be moved into a utility function?

Also, please use the term "directory" instead of "folder". "Folder" is used on windows because it displays stuff as "Folders" in the WIndows Explorer. And the [Control Panel or the Network Neighborhood are folders, but not directories](https://devblogs.microsoft.com/oldnewthing/20110216-00/?p=11473). We are referring to the file system here, so directory is the correct term. /smartass-mode off

Reply:
I have decided to keep the existing names to minimize the changes. But we could rename the variables.

Reply:
It's not that important, but it would be good to keep it in mind for new code.

Reply:
The escape character must match the following SQL query. This gets lost when moving the code into a function. Not really helpful here.

Reply:
Btw, I am not going to write any new database code in C++ ;) Only inevitable bug fixes.